rust_src/src/coding.rs

```diff
@@ -98,4 +98,16 @@ pub fn encode_file_name(fname: LispStringRef) -> LispStringRef {
     unsafe { c_encode_file_name(fname.into()) }.into()
 }
 
+/// Implements DECODE_SYSTEM macro
+/// Decode the string `input_string` using the specified coding system
+/// for system functions, if any.
+pub fn decode_system(input_string: LispStringRef) -> LispStringRef {
```

Comment:
`into()` is used in Rust to transition from one type to another aka casting. The From and Into traits are related to each other.

```
let foo = LispObject::from(1);
// and
let bar: LispObject = 1.into();
```
are equivalent. If you manually transform the calls to into() in this function you will see `LispStringRef::from(input_string)` which is already a string. Clippy points these out for you.

Reply:
ok, thank you... really need to develop "run clippy" habit :)

Reply:
Clippy can be frustrating because it runs a build. So you end up `cargo clean; cargo clippy` otherwise it fools you into believing everything is fine.